buildrunner/validation/step.py

```diff
@@ -0,0 +1,106 @@
+"""
+Copyright 2023 Adobe
+All Rights Reserved.
+
+NOTICE: Adobe permits you to use, modify, and distribute this file in accordance
+with the terms of the Adobe license agreement accompanying it.
+"""
+
+from typing import Any, Dict, List, Optional, Union
+
+# pylint: disable=no-name-in-module
+from pydantic import BaseModel, Field
+
+
+class StepPypiPush(BaseModel):
+    """ Step pypi push model"""
+    repository: str
+    username: str
+    password: str
+
+
+class Artifact(BaseModel):
+    """ Artifact model """
+    # Intentionally loose restrictions
+    format: Optional[str]
+    type: Optional[Any]
+    compression: Optional[str]
+    push: Optional[bool]
+
+
+class StepRun(BaseModel, extra='forbid'):
+    """ Run model within a step """
+
+    xfail: Optional[bool]
+    services: Optional[Dict[str, Any]]
+    image: Optional[str]
+    cmd: Optional[str]
+    cmds: Optional[List[str]]
+    # Intentionally loose restrictions
+    provisioners: Optional[Dict[str, str]]
+    shell: Optional[str]
+    cwd: Optional[str]
+    user: Optional[str]
+    hostname: Optional[str]
+    dns: Optional[List[str]]
+    dns_search: Optional[str]
+    extra_hosts: Optional[Dict[str, str]]
+    env: Optional[Dict[str, str]]
+    files: Optional[Dict[str, str]]
+    caches: Optional[Dict[str, Union[str, List[str]]]]
+    ports: Optional[Dict[str, str]]
+    volumes_from: Optional[List[str]]
+    ssh_keys: Optional[List[str]] = Field(alias='ssh-keys')
+    artifacts: Optional[Dict[str, Union[Artifact, None]]]
+    pull: Optional[bool]
+    platform: Optional[str]
+    systemd: Optional[bool]
+    cap_add: Optional[Union[str, List[str]]]
+    privileged: Optional[bool]
+    post_build: Optional[Union[str, Dict[str, Any]]] = Field(alias='post-build')
+    containers: Optional[List[str]]
+    no_cache: Optional[bool] = Field(alias='no-cache')
+
+
+class StepRemote(BaseModel, extra='forbid'):
+    """ Remote model within a step """
+    # Not sure if host is optional or required
+    host: Optional[str]
+    cmd: str
+    artifacts: Optional[Dict[str, Union[Artifact, None]]]
+
+
+class StepBuild(BaseModel, extra='forbid'):
+    """ Build model within a step """
+    path: Optional[str]
+    dockerfile: Optional[str]
+    pull: Optional[bool]
+    platform: Optional[str]
+    platforms: Optional[List[str]]
+    inject: Optional[Dict[str, str]]
```

Comment:
It *might* be possible to use inject with a key but no value, and it defaults to the working dir, but I could be remembering wrong here. It might be worth checking inject in various buildrunner files in git.

Reply:
I couldn't find a buildrunner file which had a inject value as None. But I am ok with making this check less strict. I'll make the value Optional.